drivers/png/png_driver_common.cpp

```diff
@@ -124,6 +124,51 @@ Error png_to_image(const uint8_t *p_source, size_t p_size, bool p_force_linear,
 	return OK;
 }
 
+#ifdef FPNG_ENABLED
+#include <fpng.h>
+
+Error image_to_png(const Ref<Image> &p_image, Vector<uint8_t> &p_buffer) {
+	static bool fpng_init_called = false;
+	Ref<Image> source_image = p_image->duplicate();
+
+	if (source_image->is_compressed()) {
+		source_image->decompress();
+	}
+
+	ERR_FAIL_COND_V(source_image->is_compressed(), FAILED);
+
+	// Convert to a supported format if need be
+	if (!(source_image->get_format() == Image::FORMAT_RGB8 || source_image->get_format() == Image::FORMAT_RGBA8)) {
+		if (source_image->detect_alpha()) {
+			source_image->convert(Image::FORMAT_RGBA8);
+		} else {
+			source_image->convert(Image::FORMAT_RGB8);
+		}
+	}
```

Comment:
Instead of overriding `image_to_png` and doing this, it probably should plug into existing `image_to_png` and use `fpng` only for supported formats, otherwise results of saving L8/LA8 images will be inconsistent and unexpected.